examples/transfer-files/README.md

```diff
@@ -0,0 +1,162 @@
+# Tutorial - Transfer files between the browser and other IPFS nodes
+
+> Welcome! This tutorial will help you exchange files between browser nodes and go-ipfs nodes.
+
+There are a couple of caveats:
+
+- js-ipfs currently doesn't support DHT peer discovery, the peer from which you are fetching data should be within the reach (local or in public IP) of the browser node.
+- We need to use a signalling server to establish the WebRTC connections, this won't be necessary as soon as libp2p-relay gets developed
+
+That being said, we will explain throughout this tutorial to circunvent the caveats and once they are fixed, we will update the tutorial as well.
+
+## Application diagram
+
+The goal of this tutorial is to create a application with a IPFS node that dials to other instances of it using WebRTC, and at the same time dial and transfer files from a Desktop IPFS node using WebSockets as the transport.
+
+```
+┌──────────────┐                   ┌──────────────┐
+│   Browser    │                   │   Browser    │
+│              │      WebRTC       │              │
+│              │◀─────────────────▶│              │
+└──────────────┘                   └──────────────┘
+        ▲                                  ▲
+        │                                  │
+        │WebSockets              WebSockets│
+        │        ┌──────────────┐          │
+        │        │   Desktop    │          │
+        └───────▶│              │◀─────────┘
+                 └──────────────┘
+```
+
+## Check out the final state
+
+In the end, you should get an app running, something like this:
+
+![](https://ipfs.io/ipfs/Qmbti2nBZWxQLhpggB7tC3HvcxTMivmMo3MVwQveAsHBAE)
+
+## Step-by-step instructions
+
+Here's what we are going to be doing, today:
+
+- 1. Set up, install a go-ipfs node in your machine
+- 2. Make your daemons listen on WebSockets
+- 3. Start the WebApp
+- 4. Dial to a node using WebSockets (your Desktop ones)
+- 5. Transfer files between all of your nodes, have fun!
+
+Let's go.
+
+### 1. Set up
+
+You'll need to have an implementation of IPFS running on your machine. Currently, this means either go-ipfs or js-ipfs.
+
+Installing go-ipfs can be done by installing the binary [here](https://ipfs.io/ipns/dist.ipfs.io/#go-ipfs). Alternatively, you could follow the instructions in the README at [ipfs/go-ipfs](https://github.com/ipfs/go-ipfs).
+
+Installing js-ipfs requires you to have node and [npm](https://www.npmjs.com). Then, you simply run:
+
+```sh
+> npm install --global ipfs
+...
+> jsipfs --help
+Commands:
+...
+```
+
+This will alias `jsipfs` on your machine; this is to avoid issues with `go-ipfs` being called `ipfs`.
+
+At this point, you have either js-ipfs or go-ipfs running. Now, initialize it:
+
+```sh
+> ipfs init
+# or
+> jsipfs init
+```
+
+This will set up your IPFS repo in your home directory.
+
+### 2. Make your daemons listen on WebSockets
```

Comment:
This is only applicable for Go right since JS sets this up automatically if I remember correctly?